pkgs/applications/editors/jupyter-kernels/coq/default.nix

```diff
@@ -0,0 +1,58 @@
+{ lib
+, stdenv
+, callPackage
+, runCommand
+, makeWrapper
+, coq
+, imagemagick
+, python3
+}:
```

Comment:
could you move the file to pkgs/by-name as explained here https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md ? there are plenty of examples in the latest PRs

Reply:
Hmm, I think that may not be the best idea here, for the following two reasons:

1. I think there are significant advantages to keeping the Jupyter kernels together in a common folder. In the past they've been somewhat fragmented through the codebase, and it makes it hard to refactor them together in a consistent way. I'd like to do more PRs to improve their cleanliness and ergonomics, and it may involve factoring out common code.
2. This `pkgs/by-name` effort sounds like it is for top-level packages. I wouldn't describe these kernels as top-level packages. I think we want to work toward a world where the kernels are consolidated into a common little package set and there's some nice machinery for launching notebooks/console/etc. with your desired subset of kernels. My `jupyter-console.withSingleKernel` PR was a first step in this direction. Perhaps the full set of kernels should be exposed in a top-level `jupyter-kernels` attrset, but they're not much use on their own at the top level.

Reply:
Fair enough. I kinda like to have them all in a subfolder too.